crates/store/src/db/models/queries/transactions.rs

```diff
@@ -223,66 +230,73 @@ impl TransactionSummaryRowInsert {
     }
 }
 
-/// Select complete transaction records for the given accounts and block range.
+/// Row shape: (`tx`, optional `note`, optional `note_script`)
+type TxNoteRow = (TransactionRecordRaw, Option<NoteRecordRawRow>, Option<Vec<u8>>);
+
+/// Select complete transaction records together with their emitted notes using a single join.
+///
+/// This keeps the same payload limit semantics used by [`select_transactions_records`] and
+/// avoids a second round trip to fetch notes separately.
 ///
 /// # Parameters
 /// * `account_ids`: List of account IDs to filter by
 ///     - Limit: 0 <= size <= 1000
 /// * `block_range`: Range of blocks to include inclusive
 ///
 /// # Returns
-/// A tuple of (`last_block_included`, `transaction_records`) where:
-/// - `last_block_included`: The highest block number included in the response
-/// - `transaction_records`: Vector of transaction records, limited by payload size
 ///
-/// # Note
-/// This function returns complete transaction record information including state commitments
-/// and note IDs, allowing for direct conversion to proto `TransactionRecord` without loading
-/// full block data. We use a chunked loading strategy to prevent memory exhaustion attacks and
-/// ensure predictable resource usage.
+/// A tuple containing the last block number included and a vector of
+/// [`TransactionRecordWithNotes`] types.
 ///
 /// # Raw SQL
 /// ```sql
 /// SELECT
-///     account_id,
-///     block_num,
-///     transaction_id,
-///     initial_state_commitment,
-///     final_state_commitment,
-///     input_notes,
-///     output_notes,
-///     size_in_bytes
-/// FROM
-///     transactions
-/// WHERE
-///     block_num >= ?1
-///     AND block_num <= ?2
-///     AND account_id IN (?3)
-///     AND (
-///         block_num > ?4 OR (block_num = ?4 AND transaction_id > ?5)
-///     )
+///     tx.account_id,
+///     tx.block_num,
+///     tx.transaction_id,
+///     tx.initial_state_commitment,
+///     tx.final_state_commitment,
+///     tx.nullifiers,
+///     tx.output_notes,
+///     tx.size_in_bytes,
+///
+///     n.committed_at,
+///     n.batch_index,
+///     n.note_index,
+///     n.note_id,
+///     n.note_commitment,
+///     n.note_type,
+///     n.sender,
+///     n.tag,
+///     n.aux,
+///     n.execution_hint,
+///     n.assets,
+///     n.inputs,
+///     n.serial_num,
+///     n.inclusion_path,
+///     n.script_root,
+///     s.script  -- nullable
```

Comment:
I think this might not be helpful :D Too much, though its accurate.

I wonder if maybe instead we just say.. select stuff..

Not a strong feeling; feel free to ignore.

```suggestion
///     tx.{...}
///     n.{...}
```